data_structures/k_tree/k_tree.py

```diff
@@ -0,0 +1,81 @@
+class Node:
+    def __init__(self, val, children=[]):
+        self.val = val
+        self.children = []
+        for child in children:
+            self.children.append(Node(child))
+
+    def __repr__(self):
+        return 'node is {}'.format(str(self.val))
+
+    def __str__(self):
+        return str(self.val)
+
+
+class KTree:
+    def __init__(self, val):
+        self.root = Node(val)
+
+    def __repr__(self):
+        return 'root is {}'.format(str(self.root.val))
+
+    def __str__(self):
+        return str(self.root.val)
+
+    def insert(self, val, parent):
+        self.breadth_first_traversal(lambda x: x.children.append(Node(val)) if x.val == parent else False)
+
+    def in_order(self, operation):
+        def _walk(node=None):
+            if node is None:
+                return
+
+            if node.left is not None:
+                _walk(node.left)
+
+            operation(node)
+
+            if node.right is not None:
+                _walk(node.right)
+
+        _walk(self.root)
+
+    def pre_order(self, operation):
+        def _walk(node=None):
+            if node is None:
+                return
+
+            operation(node)
+
+            if len(node.children):
```

Comment:
You actually do not need to check that the node has children since the `for` loop does not execute if there are no items in the list.